### PR TITLE
GH action for CI validation

### PR DESCRIPTION
**Pull request by cbbayburt**
Add CI validation workflows for terraform and Salt.

Salt validation runs a `show_highstate` to ensure proper rendering for various roles, using test values for all the necessary grains.

Terraform validation validates, applies and destroys configurations based on the `main.tf.*.example` files using the 'null' backend.

These validations replace deprecated Travis jobs.

See some test runs in action: https://github.com/cbbayburt/sumaform/actions/workflows/ci-validation.yml

Fixes https://github.com/SUSE/spacewalk/issues/21505

### CODE REVIEW

File: .github/workflows/ci-validation/terraform-validation [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+# Remove providers to use the 'null' provider
+sed -i '/provider *"/,/\}/d' main.tf.*.example
+# Remove libvirt provider dependency
+sed -i '/libvirt = {/,/\}/d' main.tf.*.example
+
+ln -sfn ../backend_modules/null modules/backend
+
+# validate examples
+for tf_file in `ls main.tf.*.example`; do
+  echo "::notice::Validating $tf_file..."
```
Comment (nodeg) [inline .github/workflows/ci-validation/terraform-validation:16]: See https://www.shellcheck.net/wiki/SC2045

File: modules/cucumber_testsuite/main.tf [hunk 2]
```diff
@@ -25,7 +25,7 @@ locals {
   additional_repos          = { for host_key in local.hosts :
     host_key => lookup(var.host_settings[host_key], "additional_repos", {}) if var.host_settings[host_key] != null }
   additional_repos_only     = { for host_key in local.hosts :
-    host_key => lookup(var.host_settings[host_key], "additional_repos_only", {}) if var.host_settings[host_key] != null }
+    host_key => lookup(var.host_settings[host_key], "additional_repos_only", false) if var.host_settings[host_key] != null }
```
Comment (nodeg) [inline modules/cucumber_testsuite/main.tf:28]: Is this one of the bugs you mentioned you found during implementing this?

**Review by nodeg — APPROVED** [review]
LGTM. Thank you Can! I left some small comments/suggestions.

File: .github/workflows/ci-validation.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
+name: CI validation tests
+
+on:
+  push: # For testing only
+    branches:
+      - deployment-tests
```
[the remark this answers is not in the document]
Comment (cbbayburt) [reply .github/workflows/ci-validation.yml:6]: I think the pull_request event would be enough for our case. We want everyone to push via PRs anyway. Experimenting in such a branch might eat up too much GH runner bandwidth.

File: modules/cucumber_testsuite/main.tf (hunk 2 above)
Comment (cbbayburt) [reply modules/cucumber_testsuite/main.tf:28]: Yes, `additional_repos_only` was supposed to be a `boolean`

File: .github/workflows/ci-validation/terraform-validation (hunk 1 above)
Comment (cbbayburt) [reply .github/workflows/ci-validation/terraform-validation:16]: I had no idea this website exists. Thanks :D

Comment (nodeg) [reply .github/workflows/ci-validation/terraform-validation]: You're welcome. There are shellcheck plugins for nearly all popular IDEs/editors available. 
For VS Code for example: https://github.com/vscode-shellcheck/vscode-shellcheck

File: .github/workflows/ci-validation.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (nodeg) [reply .github/workflows/ci-validation.yml:6]: Fine with me.

**Pull request merged**